### PR TITLE
fotoxx: Update to 20.04 ; Fix musl build

### PR DESCRIPTION
**Pull request by ndowens**

### CODE REVIEW

File: srcpkgs/fotoxx/template [hunk 1]
```diff
@@ -13,17 +13,25 @@ maintainer="Orphaned <orphan@voidlinux.org>"
...
+	musl)
+		makedepends+=" libexecinfo-devel";;
+esac
+
+pre_build() {
+case "$XBPS_TARGET_LIBC" in
+	musl)
+		patch -Np0 < "$FILESDIR"/libexecinfo.patch;;
+esac
+}
```
Comment (xtraeme) [inline srcpkgs/fotoxx/template]: please don't add such patches based on a condition

**Comment by ndowens** [comment]
I tried without, but on GLIBC systems, it does not know -lexecinfo.

File: srcpkgs/fotoxx/template (hunk 1 above)
Comment (ndowens) [reply srcpkgs/fotoxx/template]: Found an alternative way

**Comment by ndowens** [comment]
@xtraeme Can you relook at this please :)

**Comment by pullmoll** [comment]
@ndowens If you change the Makefile to add to LIBS, i.e. `LIBS += ` instead of `LIBS = `, you can then just set `LIBS="-lexecinfo"` in the template and avoid the `EXECINFO` variable which otherwise pollutes the environment.

**Comment by ndowens** [comment]
> @ndowens If you change the Makefile to add to LIBS, i.e. `LIBS += ` instead of `LIBS = `, you can then just set `LIBS="-lexecinfo"` in the template and avoid the `EXECINFO` variable which otherwise pollutes the environment.

Will try that at some point today.

**Comment by ndowens** [comment]
Ok, I set LIBS = to LIBS += in Makefile and added LIBS="-lexecinfo" but that does not add -lexecinfo into LIBS in Makefile

Also tried doing make_build_args="LIBS=lexecinfo" and that caused way more link errors, as that overwrote the LIBS.

File: srcpkgs/fotoxx/patches/execinfo.patch [hunk 2]
```diff
@@ -0,0 +1,17 @@
+diff --git Makefile Makefile
+index 218aa2d..77ec75e 100644
+--- Makefile
++++ Makefile
+@@ -47,4 +47,4 @@ CFLAGS = $(CXXFLAGS) $(CPPFLAGS) -c                \
+    -I/usr/include/libchamplain-0.12/
+
+-LIBS = `$(PKG_CONFIG) --libs gtk+-3.0`                                              \
++LIBS = `$(PKG_CONFIG) --libs gtk+-3.0` $(shell pkg-config --libs libexecinfo)    \
```
Comment (xtraeme) [inline srcpkgs/fotoxx/patches/execinfo.patch]: You should modify `Makefile.am` instead, and then you'll be able to submit it to upstream.

For this you'll need to add `automake` in hostmakedepends and run `autoreconf -fi` at pre_configure().

Comment (ndowens) [reply srcpkgs/fotoxx/patches/execinfo.patch]: Will try this ASAP

Comment (xtraeme) [reply srcpkgs/fotoxx/patches/execinfo.patch]: hmm ok

**Comment by pullmoll** [comment]
Since adding the pkg-config file makes more sense, correcting my hint is redundant.
The problem was/is that `common/build-style/gnu-makefile` does not pass `LIBS="$LIBS"` in `do_build()`. It should probably do that for similar cases for when `make_use_env` is defined.

You could also have tried to set `make_build_args="LIBS=-lexecinfo`` in the template.

Adding the library like `LDFLAGS="-lexecinfo"` should also have worked while IMO this is not good style because `LIBS` is meant to be used for adding libraries, not `LDFLAGS`.

**Comment by ndowens** [comment]
Yea I tried make_build_args but it made it much worse

**Pull request merged**